src/utils.ts

```diff
@@ -46,16 +49,15 @@ export class Utils {
         return obj.hasOwnProperty(prop);
     }
 
-    public static parseTimestampsAsDates(obj: unknown): unknown {
-        if (
-            typeof obj === "object" &&
-            obj !== null &&
-            this.hasOwnProperty(obj, "Timestamp") &&
-            typeof obj.Timestamp === "string"
-        ) {
-            obj.Timestamp = new Date(obj.Timestamp);
+    public static parseTimestampsAsDates(obj: unknown): void {
+        if (typeof obj === "object" && obj != null) {
+            if (this.hasOwnProperty(obj, "Timestamp") && typeof obj.Timestamp === "string") {
+                obj.Timestamp = new Date(obj.Timestamp);
+            }
+            Object.values(obj).map((value) => {
+                this.parseTimestampsAsDates(value);
+            });

```

Comment:
Just checking: is it `safe` to access for `.values` even when `obj` is not an array?

Reply:
Are you asking safe from a programming language perspective or in the sense that there may be `Timestamp` properties inside non-array objects that we don't want to mess with?

If it's the former, then yes, it's safe. I guess the main benefit of `Object.values` is to loop over values of a non-array object, otherwise you could loop over the elements of the array itself 😛 

If it's the latter, then I'm not sure. But is there a context in which we _want_ a `Timestamp` property to be a string over a Date?

Reply:
I meant the former :) . I'm just trying to make sure we can always call `Object.values` without causing some exception, which sounds to be the case.

Reply:
Yep! At least that's my understanding. If we call `Object.values` on an array, we get the same array back, and if we call it on an object it returns a list of values:

```node
> Object.values([1,2,3])
[ 1, 2, 3 ]
> Object.values({a:1, b:2, c:3})
[ 1, 2, 3 ]
```

I think the only problematic scenario would be calling `Object.values` on `null` or `undefined`, but that should be protected against here